### PR TITLE
feat(risedev): introduce sql server to risedev

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as titled.

An example:
```
      - use: sqlserver
        # user-managed: true
        address: localhost
        port: 1433
        user: SA
        password: "YourPassword123"
        database: "master"
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/risedevtool/src/task/sql_server_service.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self.user_managed
+    }
+
+    fn image(&self) -> String {
+        self.image.clone()
+    }
+
+    fn envs(&self) -> Vec<(String, String)> {
+        vec![
+            ("ACCEPT_EULA".to_owned(), "Y".to_owned()),
```
Comment (xxchan) [inline src/risedevtool/src/task/sql_server_service.rs:33]: > By passing the value "Y" to the environment variable "ACCEPT_EULA", you are expressing that you have a valid and existing license for the edition and version of SQL Server that you intend to use. You also agree that your use of SQL Server software running in a Docker container image will be governed by the terms of your SQL Server license.

https://arc.net/l/quote/uetmwpuy

🤔 Looks dangerous

Comment (xxchan) [reply src/risedevtool/src/task/sql_server_service.rs:33]: Oh, developer edition (the default) looks OK.

> SQL Server Developer edition lets developers build any kind of application on top of SQL Server. It includes all the functionality of Enterprise edition, but is licensed for use as a development and test system, not as a production server.

File: src/risedevtool/src/risedev_env.rs [hunk 2]
```diff
@@ -118,6 +118,19 @@ pub fn generate_risedev_env(services: &Vec<ServiceConfig>) -> String {
...
+                let user = &c.user;
+                let password = &c.password;
+                let database = &c.database;
+                // These envs are used by `sqlcmd`.
+                writeln!(env, r#"SQLCMDSERVER="{host}""#,).unwrap();
+                writeln!(env, r#"SQLCMDPORT="{port}""#,).unwrap();
+                writeln!(env, r#"SQLCMDUSER="{user}""#,).unwrap();
+                writeln!(env, r#"SQLCMDPASSWORD="{password}""#,).unwrap();
+                writeln!(env, r#"SQLCMDDBNAME="{database}""#,).unwrap();
+            }
```
Comment (xxchan) [inline src/risedevtool/src/risedev_env.rs:133]: What about adding a `RISEDEV_SQLSERVER_WITH_OPTIONS_COMMON`?

**Review by xxchan — COMMENTED** [review]
LGTM. Can we change the e2e tests to it: Add to `ci-inline-source-test` profile, and only let this section to handle it.
 https://github.com/risingwavelabs/risingwave/blob/63801dd9a14d15f83812c04212ae3c718ff9b423/ci/scripts/e2e-source-test.sh#L38-L41

File: src/risedevtool/src/risedev_env.rs (hunk 2 above)
Comment (KeXiangWang) [reply src/risedevtool/src/risedev_env.rs:138]: added

**Comment by KeXiangWang** [comment]
> LGTM. Can we change the e2e tests to use risedev? i.e., add to ci-inline-source-test profile, and only let this section to handle it.

Do you mean moving cdc source tests to `e2e_test/source_inline`?

**Comment by xxchan** [comment]
> > LGTM. Can we change the e2e tests to use risedev? i.e., add to ci-inline-source-test profile, and only let this section to handle it.
> 
> Do you mean moving cdc source tests to `e2e_test/source_inline`?

Yes. Also remove `export` in `e2e-source-test.sh`

**Comment by KeXiangWang** [comment]
> Yes. Also remove export in e2e-source-test.sh

Currently, the cdc source tests include recovery tests. When we support recovery in `slt`, we can do it and it should be a separate PR.

**Review by xxchan — APPROVED** [review]

**Pull request merged**